nova/virt/vmwareapi/vmops.py

```diff
@@ -2050,12 +2059,22 @@ def finish_migration(self, context, migration, instance, disk_info,
             reattach_volumes = True
             LOG.debug("Relocating VM for migration to %s",
                       migration.dest_compute, instance=instance)
-            self._relocate_vm(vm_ref, context, instance, network_info,
-                              image_meta)
-            LOG.debug("Relocated VM to %s", migration.dest_compute,
-                      instance=instance)
-            vm_util.update_cluster_placement(self._session, context,
-                                             instance, self._cluster, vm_ref)
+            try:
+                self._relocate_vm(vm_ref, context, instance, network_info,
+                                  image_meta)
+                LOG.debug("Relocated VM to %s", migration.dest_compute,
+                          instance=instance)
+            except Exception as e:
+                with excutils.save_and_reraise_exception():
+                    LOG.error("Relocating the VM failed with error: %s", e,
+                              instance=instance)
+                    self._attach_volumes(instance, block_device_info)
+            else:
+                vm_util.update_cluster_placement(self._session,
```

Comment:
We don't need the `else` here, I think, because we don't have a `finally` and the `except` will re-raise the exception.